relay-event-schema/src/protocol/event.rs

```diff
@@ -65,6 +65,16 @@ impl FromStr for EventId {
 
 relay_common::impl_str_serde!(EventId, "an event identifier");
 
+impl TryFrom<&SpanId> for EventId {
+    type Error = ();
```

Comment:
We already have `thiserror` could make a quick proper error for this:

```rs
#[derive(Debug, thiserror::Error)]
#[error("cannot convert span id to event id")]
pub struct TryFromSpanIdError;
```

Could also include the source error `pub struct TryFromSpanIdError(FooError);`

Reply:
I decided to make this consistent with the `FromStr for EventId` implementation above, but fleshed out the error type in `TryFrom<&Span>` a bit.